### PR TITLE
Datastore discrimination ref docs and sample

### LINKED ISSUE #1475: Inheritance hierarchy for a single kind
Is it possible to support inheritance strategy for a single kind which containing entities for two or more different classes in an inheritance hierarchy?

For example, we have types:
```

public class Shape {    
    private String id;
    //other fields
}

public class Circle extends Shape {}

public class Polygon extends Shape{}

```
I would like to get information about all the shapes which was saved.

```
var circe = new Circle();
var polygon = new Polygon();
shapeRepository.save(circle);
shapeRepository.save(polygon);
var shapes = shapeRepository.findAll();

```

Right now I receive a collection with 2 elements of 'Shape' type. Is it possible to add mechanisms, for example like [JPA discriminators](https://docs.oracle.com/javaee/7/api/javax/persistence/DiscriminatorColumn.html), which will make childs (which are in an inheritance hierarchy) into their real types?

### PR DESCRIPTION
**Pull request by ChengyuanZhao**
fixes #1475 
refdoc and sample update for discrimination

### CODE REVIEW

File: docs/src/main/asciidoc/datastore.adoc [hunk 1]
```diff
@@ -351,6 +351,59 @@ public class ConverterConfiguration {
...
+@DiscriminationValue("pug")
+class Pug extends Dog {
+	@Override
+	String speak() {
+		return "woof woof";
+	}
+}
+----
+
+Instances of all 4 types are stored in the `pets` Kind.
```
Comment (dmitry-s) [inline docs/src/main/asciidoc/datastore.adoc]: Pet is an abstract class, so only 3 types are stored in the kind.

File: docs/src/main/asciidoc/datastore.adoc [hunk 2]
```diff
@@ -351,6 +351,59 @@ public class ConverterConfiguration {
...
+Because a single Kind is used, all classes in the hierarchy must share the same ID property and no two instances of any type in the hierarchy can share the same ID value.
+
+Entity rows in Cloud Datastore store their respective types' `DiscriminationValue` in a field specified by the root superclass's `DiscriminationField` (`pet_type` in this case).
+Reads and queries using a given type parameter will match each entity with its specific type.
+For example, reading a `List<Pet>` will produce a list containing instances of all 4 types.
```
Comment (dmitry-s) [inline docs/src/main/asciidoc/datastore.adoc]: same here

File: docs/src/main/asciidoc/datastore.adoc (hunk 1 above)
Comment (ChengyuanZhao) [reply docs/src/main/asciidoc/datastore.adoc]: fixed !

**Comment by meltsufin** [comment]
@ChengyuanZhao @dmitry-s Can we please change `@DiscriminationField` and `@DiscriminationValue` to `@DiscriminatorField` and `@DiscriminatorValue` more in line with how these annotations are called in JPA?

**Pull request merged**